### PR TITLE
Minor TSan documentation improvements.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Minor TSan documentation improvements to improve the readability and one clarification of the behaviour to TSan around tracking memory accesses.  

cc @OlivierNicole

### CODE REVIEW

File: manual/src/cmds/tsan.etex [hunk 1]
```diff
@@ -137,13 +137,13 @@ increased by a factor between 4 and 7.
 
 As illustrated by the previous example, TSan will only report the data races
 encountered during execution. Another important caveat is that TSan remembers
-only a finite number of memory accesses per memory location and per thread. At
+only a finite number of memory accesses per memory location. At
```
Comment (gasche) [inline manual/src/cmds/tsan.etex:140]: I think that the meaning changed here, this is not only a grammar fix. Could you preserve the previous meaning which I think is the correct one?

Comment (OlivierNicole) [reply manual/src/cmds/tsan.etex:140]: Actually I am the one who asked @tmcgilchrist to include this in the PR (and the one to blame for the error). Memory accesses are remembered globally.

**Review by OlivierNicole — APPROVED** [review]
This looks good to me. It fixes an erroneous statement and a number of typos.

File: manual/src/cmds/tsan.etex (hunk 1 above)
Comment (tmcgilchrist) [reply manual/src/cmds/tsan.etex:140]: The additional information `per thread` is incorrect/misleading. Memory accesses are tracked globally with a maximum of 4 memory accesses tracked. My understanding is based off the described [Thread Sanitizer State Machine Algorithm](https://github.com/google/sanitizers/wiki/ThreadSanitizerAlgorithm#state-machine) and the count of shadow values in a shadow cell from llvm [here](https://github.com/llvm/llvm-project/blob/release/15.x/compiler-rt/lib/tsan/rtl/tsan_defs.h#L76-L77) and gcc [here](https://github.com/gcc-mirror/gcc/blob/releases/gcc-12.3.0/libsanitizer/tsan/tsan_defs.h#L103-L104). Which looks consistent across the various releases.

**Review by gasche — APPROVED** [review]

File: manual/src/cmds/tsan.etex (hunk 1 above)
Comment (gasche) [reply manual/src/cmds/tsan.etex:140]: Thanks for the clarification!

**Pull request merged**